.github/workflows/format_lint_test.yaml

```diff
@@ -4,52 +4,65 @@ on:
     branches: [master, develop]
 
 env:
-  FLUTTER_VERSION: 3.13.7
   JAVA_VERSION: 17.x
 
 jobs:
   format:
-    name: "Format"
+    name: Format
     runs-on: ubuntu-latest
     defaults:
       run:
         working-directory: ./uni
     steps:
-      - uses: actions/checkout@v4
-      - uses: subosito/flutter-action@v2
+      - name: Clone repository
+        uses: actions/checkout@v4
+
+      - uses: mikefarah/yq@master
+        name: Get Flutter version
+        id: get_flutter_version
         with:
-          flutter-version: ${{ env.FLUTTER_VERSION }}
+          cmd: yq '.environment.flutter' pubspec.yaml
+
+      - name: Set up Flutter
+        uses: subosito/flutter-action@v2
+        with:
+          channel: stable
+          flutter-version: ${{ steps.get_flutter_version.outputs.result }}
 
       - run: dart format $(find . -type f -name "*.dart" -a -not -name "*.g.dart" -a -not -name "*.mocks.dart") --set-exit-if-changed
 
   lint:
-    name: "Lint"
+    name: Lint
     runs-on: ubuntu-latest
     needs: format
     defaults:
       run:
         working-directory: ./uni
     steps:
-      - uses: actions/checkout@v4
+      - name: Clone repository
+        uses: actions/checkout@v4
+
       - uses: actions/setup-java@v3
         with:
           java-version: ${{ env.JAVA_VERSION }}
-          distribution: "zulu"
-      - uses: subosito/flutter-action@v2
-        with:
-          flutter-version: ${{ env.FLUTTER_VERSION }}
+          distribution: zulu
 
-      - name: Cache pub dependencies
-        uses: actions/cache@v3
+      - uses: mikefarah/yq@master
+        name: Get Flutter version
+        id: get_flutter_version
+        with:
+          cmd: yq '.environment.flutter' pubspec.yaml
+  
+      - name: Set up Flutter
+        uses: subosito/flutter-action@v2
         with:
-          path: ${{ env.FLUTTER_HOME }}/.pub-cache
-          key: ${{ runner.os }}-pub-${{ github.ref }}-${{ hashFiles('**/pubspec.lock') }}
-          restore-keys: ${{ runner.os }}-pub-${{ github.ref }}-
```

Comment:
This means we are no longer caching dependencies, right?
From what I understand, we're not using them anywhere else, but maybe we should start using them in all actions... subosito/flutter-action seems to support caching pub dependencies (https://github.com/subosito/flutter-action)

Reply:
Yeah, this is a good idea. But given that we get CI for free anyway (hehe), I often tend not to care about cache.

I'll do it.